mp4parse/src/lib.rs

```diff
@@ -709,6 +709,9 @@ fn read_moov<T: Read>(f: &mut BMFFBox<T>, context: &mut MediaContext) -> Result<
 }
 
 fn read_pssh<T: Read>(src: &mut BMFFBox<T>) -> Result<ProtectionSystemSpecificHeaderBox> {
+    if src.head.size > BUF_SIZE_LIMIT as u64 {
+        return Err(Error::InvalidData("Invalid pssh data."));
+    }
     let mut box_content = Vec::with_capacity(src.head.size as usize);
     src.read_to_end(&mut box_content)?;
```

Comment:
This pattern should be using read_buf instead, which already handles this check.

Reply:
Yes, thanks for the suggestion.